src/vue/modules/assets-explorer/components/AssetCard.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <a class="asset-card" v-bind="$listeners">
+    <div class="asset-card__header">
+      <logo-viewer class="asset-card__logo" :asset="asset" />
+    </div>
+    <div class="asset-card__info">
+      <p class="asset-card__code">
+        {{ asset.code }}
+      </p>
+      <p class="asset-card__name">
+        {{ asset.name }}
+      </p>
+      <p v-if="balance" class="asset-card__balance">
+        {{
+          'assets-page.list-item-balance-line' | globalize({
+            value: balance.available
+          })
+        }}
+      </p>
+      <p v-else class="asset-card__balance asset-card__no-balance">
+        {{ 'assets-page.no-balance-msg' | globalize }}
+      </p>
+    </div>
+  </a>
+</template>
+
+<script>
+import LogoViewer from './LogoViewer'
+
+import { AssetRecord } from '../asset-record'
+import { mapGetters } from 'vuex'
+import { types } from '../store/types'
+
+export default {
+  name: 'asset-card',
+  components: { LogoViewer },
+  props: {
+    asset: {
+      type: AssetRecord,
+      required: true,
+    },
+  },
+  computed: {
+    ...mapGetters({
+      balances: types.balances,
+    }),
+    balance () {
+      return this.balances[this.asset.code]
```

Comment:
The current singletone module implementation is a great stuff, we can move all this store-based logic to the vuex module.

The point is that AssetCard.vue does not know the balances structure, and vuex does know. So we can define a vuex getter that accept arg (I believe I thought such feasibility earlier) and call it here. If all the data-extracting data-setting logic that related to vuex-stored data would be placed in one place the code is much clearer and controllable.